tests/metrics/test_metrics.py

```diff
@@ -728,3 +729,27 @@ def test_get_course_num_learners_completed_for_time_period(self):
             end_date=self.data_end_date,
             course_id=self.alpha_course_overview.id)
         assert actual == expected
+
+
+@pytest.mark.django_db
+class TestGetLastMonthSiteMetrics(object):
+    """
+    This test also exercises the time period getters used in
+    figures.metrics.get_last_month_site_metrics
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, db):
+        self.site = SiteFactory(domain='alpha.site')
+        self.expected_keys = [
+            'active_users',
+            'site_courses',
+            'course_enrollments',
+            'registered_users',
+            'new_users',
+            'course_completions'
+        ]
+
+    def test_get_last_month_site_metrics(self):
+        data = get_last_month_site_metrics(self.site)
+        assert data.keys() == self.expected_keys
```

Comment:
This is not going to fail in any case. I think the test should validate some data.

Reply:
It's returning some data and test should test expected data. What's the point of test if it's not going to fail?

Reply:
You can mock internal functions and mold the return value of internal functions.

Reply:
I will update the unit test

Reply:
@iam-mhaseeb Unit tests modified. I have testing one of the analytics returns by function(since we already have separate function testing analytics preparations)

Reply:
@asadmanzoor93 You function returns last_month_site_metrics so it should test that instead of just one component of that metrics.

Reply:
@iam-mhaseeb as I mentioned in the above comment.. the function returns the following metrics for last month using existing functions: `active_users`, `registered_users`, `new_users`, `site_courses`, `course_enrollments`, `course_completions`.

We already have Unit tests to tests these analytics individually with time period options. It can be found here: https://github.com/edly-io/figures/blob/492434878c1a1a0e034b469147c4ebf160ebbeff/tests/metrics/test_metrics.py

Since there is no explicitly change happening in this function as from calling other functions to gather stats(for which tests are already written) so I have written as test verification for one of the main user registration instead again repeating the same tests again.